src/main/java/io/vertx/blueprint/todolist/Application.java

```diff
@@ -0,0 +1,79 @@
+package io.vertx.blueprint.todolist;
+
+import io.vertx.blueprint.todolist.service.JdbcTodoService;
+import io.vertx.blueprint.todolist.service.RedisTodoService;
+import io.vertx.blueprint.todolist.verticles.SingleApplicationVerticle;
+import io.vertx.blueprint.todolist.verticles.TodoVerticle;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.Verticle;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.redis.RedisOptions;
+
+/**
+ * Vert.x Blueprint Application - Todo Backend
+ * Launcher Class
+ *
+ * @author Eric Zhao
+ */
+public class Application {
+
+  public static Verticle RedisTodo() {
+    RedisOptions config;
+    // this is for OpenShift Redis Cartridge
+    String osPort = System.getenv("OPENSHIFT_REDIS_PORT");
+    String osHost = System.getenv("OPENSHIFT_REDIS_HOST");
+    if (osPort != null && osHost != null)
+      config = new RedisOptions()
+        .setHost(osHost).setPort(Integer.parseInt(osPort));
+    else
+      config = new RedisOptions().setHost("127.0.0.1");
+
+    return new TodoVerticle(new RedisTodoService(config));
+  }
+
+  public static Verticle JdbcTodo() {
+    // in this example we use MySQL
+    JsonObject config = new JsonObject()
```

Comment:
maybe read this from a config.json. If you use the launcher then it becomes easy to get it